dotfiles.py

```diff
@@ -1,29 +1,52 @@
 #!/usr/bin/env python3.8
-"""Module docstring."""
+
 import filecmp
+from collections import namedtuple
+from os.path import exists
 from os.path import expanduser
 from shutil import copyfile
 from typing import List
-from typing import Tuple
+from typing import NamedTuple
 
 import click
 
-DOTFILES: List[Tuple[str, str]] = [("~/.tmux.conf", "tmux.conf"), ("~/.vimrc", "vimrc")]
+
+class Dotfile(NamedTuple):
+    system_path: str
+    repo_name: str
+
+
+DOTFILES: List[Dotfile] = [
+    Dotfile(system_path=expanduser("~/.tmux.conf"), repo_name="tmux.conf"),
+    Dotfile(system_path=expanduser("~/.vimrc"), repo_name="vimrc"),
+]
 
 
 @click.group()
 def main():
-    """Pass"""
+    pass
 
 
 @main.command()
 def deploy():
-    """Deploy the dotfiles."""
-    dotfiles: Tuple[str, str] = [(expanduser(df[0]), df[1]) for df in DOTFILES]
+    """Deploy DOTFILES to system.
+
+    If the `system_path` of the dotfile already exists and matches the repo version,
+    do nothing. Otherwise, copy the repo config to the system path, overwriting
+    anything that might be there.
 
-    for system_file, my_file in dotfiles:
-        click.secho(system_file)
-        if not filecmp.cmp(my_file, system_file):
-            copyfile(my_file, system_file)
+    """
+    for dotfile in DOTFILES:
```

Comment:
These dotfiles are `namedtuple`s, so you can also destructure them like you would with a normal `tuple`. That is, instead of writing
```python
for dotfile in DOTFILES:
    do_something(dotfile.system, dotfile.repo)
```
and using the `dotfile.attr` everywhere you can also write
```python
for (system, repo) in DOTFILES:
    do_something(system, repo)
```
if you so choose.